src/common/modal.js

```diff
@@ -95,43 +77,106 @@ class Confirm extends Component {
         {body}
       </Body>
       <Footer>
-        <Button
-          btnStyle='primary'
-          onClick={this._resolve}
-          style={this._style}
-        >
-          {okLabel}
-        </Button>
-        {' '}
-        <Button
-          onClick={this._reject}
-        >
-          {cancelLabel}
-        </Button>
+        {map(buttons, ({
+          label,
+          tooltip,
+          value,
+          icon,
+          ...props
+        }) => {
+          const button = <Button
+            onClick={() => this._resolve(value)}
+            key={value}
+            {...props}
+          >
+            {icon !== undefined && <Icon icon={icon} fixedWidth />}
+            {label}
+          </Button>
+          return <span>
+            {tooltip !== undefined
+              ? <Tooltip content={tooltip}>{button}</Tooltip>
+              : button
+            }
+            {' '}
+          </span>
+        })}
+        {this.props.reject !== undefined &&
+          <Button onClick={this._reject} >
+            {_('genericCancel')}
+          </Button>
+        }
       </Footer>
     </div>
   }
 }
 
+const ALERT_BUTTONS = [ { label: _('alertOk'), value: 'ok' } ]
+
+export const alert = (title, body) => (
+  new Promise(resolve => {
+    modal(
+      <GenericModal
+        buttons={ALERT_BUTTONS}
+        resolve={resolve}
+        title={title}
+      >
+        {body}
+      </GenericModal>
+    )
+  })
+)
+
+const _addRef = (component, ref) => {
+  if (isString(component) || isArray(component)) {
+    return component
+  }
+
+  try {
+    return cloneElement(component, { ref })
+  } catch (_) {} // Stateless component.
+  return component
+}
+
+const CONFIRM_BUTTONS = [ { btnStyle: 'primary', label: _('confirmOk') } ]
+
 export const confirm = ({
   body,
-  title,
-  okLabel,
-  cancelLabel,
```

Comment:
The label params were previously used, have you updated the code to use `chooseAction` instead?